### PR TITLE
Make test for no PrivateHeaders problem

### PR DESCRIPTION
**Pull request by dpgraham**
* Looks like when using electron-builder with Node 9 and building assets to the `release/` folder, PrivateHeaders does not get transferred
* Added a test that verifies the `PrivateHeaders` folder is there so that this does not occur in the future
* Also, edited the scripts in package.json so that all of the tests are run, node modules are deleted and installed and removed a redundancy

### CODE REVIEW

File: test/e2e/main-e2e.test.js [hunk 1]
```diff
@@ -60,4 +61,13 @@ describe('application launch', function () {
     await client.pause(500);
     await client.getWindowCount().should.eventually.equal(initialWindowCount + 1);
   });
+
+  it('check that WebDriverAgent private headers folder (regression test for https://github.com/appium/appium-desktop/issues/417)', async function () {
+    // NOTE: This isn't really an "e2e" test, but the test has to be written here because the /release 
+    // folder needs to be built in order to run the test
+    if (platform === 'darwin') {
```
Comment (imurchie) [inline test/e2e/main-e2e.test.js]: I would rather the logic be to skip the test if it is on a system that is not Mac.
```js
if (platform === 'darwin') {
  return this.skip();
}
// now do the test
```

**Comment by dpgraham** [comment]
@imurchie Maybe. Do you have any ideas on how to test this?

File: test/e2e/main-e2e.test.js [hunk 2]
```diff
@@ -60,4 +61,15 @@ describe('application launch', function () {
     await client.pause(500);
     await client.getWindowCount().should.eventually.equal(initialWindowCount + 1);
   });
+
+  it('check that WebDriverAgent private headers folder (regression test for https://github.com/appium/appium-desktop/issues/417)', async function () {
```
Comment (jlipps) [inline test/e2e/main-e2e.test.js]: check that the folder does what

File: test/e2e/main-e2e.test.js [hunk 3]
```diff
@@ -60,4 +61,15 @@ describe('application launch', function () {
...
+    // NOTE: This isn't really an "e2e" test, but the test has to be written here because the /release 
+    // folder needs to be built in order to run the test
+    if (platform === 'darwin') {
```
Comment (jlipps) [inline test/e2e/main-e2e.test.js]: isn't this logic backwards?

Comment (imurchie) [reply test/e2e/main-e2e.test.js]: Yes. @dpgraham just copied-and-pasted my copy-and-paste example.

File: test/e2e/main-e2e.test.js [hunk 4]
```diff
@@ -60,4 +61,15 @@ describe('application launch', function () {
...
+      this.skip();
```
Comment (imurchie) [inline test/e2e/main-e2e.test.js]: Needs a return.

**Comment by imurchie** [comment]
@dpgraham Maybe get a hash of the original WDA folder and compare it to the one produced?

**Comment by dpgraham** [comment]
Fixes pushed

**Pull request merged**